plugins/flytekit-kf-pytorch/tests/test_elastic_task.py

```diff
@@ -65,3 +66,32 @@ def wf(config: Config = Config()) -> typing.Tuple[str, Config, torch.nn.Module,
     only be obtained if the distributed process group is initialized correctly.
     """
     assert distributed_result == sum([5 + 2 * rank + world_size for rank in range(world_size)])
+
+
+@pytest.mark.parametrize(
+    "start_method,target_exec_id,monkeypatch_exec_id_env_var",
+    [
+        ("spawn", "", False),
+        ("spawn", "f12345678", True),
+        ("fork", "local", False),
```

Comment:
When spawning, the `execution_id.name`, `.project`, `.domain`, ... are set to the default value `""` [here](https://github.com/flyteorg/flytekit/blob/c8433ea2d2ecf362815b1ee63554549ced9acd75/flytekit/bin/entrypoint.py#L203) when the `FLYTE_INTERNAL_EXECUTION_ID`, ... env vars are not set, i.e. during a local execution.

When executing a workflow/task locally, these execution identifiers are normally set to `"local"` which happens [here](https://github.com/flyteorg/flytekit/blob/c8433ea2d2ecf362815b1ee63554549ced9acd75/flytekit/core/context_manager.py#L840). Since the parent processes stack is copied during forking, `"local"` is set when using this start method.

This difference between forking and spawning in a local execution is something that gives me a bit of grief.

I see two options for removing this difference.

* Not set `""` as the default value for execution id name, project, domain, ... in `flytekit.bin.entrypoint.setup_execution`. Would this have any undesired effect?
* Maintain an adapted copy of `setup_execution` here, which would, however, lead to quite some code duplication which wouldn't be nice either.